uspace/lib/c/include/async.h

```diff
@@ -356,93 +296,32 @@ extern void async_sess_args_set(async_sess_t *, iface_t, sysarg_t, sysarg_t);
  * User-friendly wrappers for async_share_in_start().
  */
 
-#define async_share_in_start_0_0(exch, size, dst) \
-	async_share_in_start(exch, size, 0, NULL, dst)
-#define async_share_in_start_0_1(exch, size, flags, dst) \
-	async_share_in_start(exch, size, 0, flags, dst)
-#define async_share_in_start_1_0(exch, size, arg, dst) \
-	async_share_in_start(exch, size, arg, NULL, dst)
-#define async_share_in_start_1_1(exch, size, arg, flags, dst) \
-	async_share_in_start(exch, size, arg, flags, dst)
-
-extern errno_t async_share_in_start(async_exch_t *, size_t, sysarg_t,
+extern errno_t async_share_in_start_0_0(async_exch_t *, size_t, void **);
+extern errno_t async_share_in_start_0_1(async_exch_t *, size_t,
     unsigned int *, void **);
+extern errno_t async_share_in_start_1_0(async_exch_t *, size_t, sysarg_t,
+    void **);
+extern errno_t async_share_in_start_1_1(async_exch_t *, size_t, sysarg_t,
+    unsigned int *, void **);
+
 extern bool async_share_in_receive(ipc_call_t *, size_t *);
 extern errno_t async_share_in_finalize(ipc_call_t *, void *, unsigned int);
 
 extern errno_t async_share_out_start(async_exch_t *, void *, unsigned int);
 extern bool async_share_out_receive(ipc_call_t *, size_t *, unsigned int *);
 extern errno_t async_share_out_finalize(ipc_call_t *, void **);
 
-/*
- * User-friendly wrappers for async_data_read_forward_fast().
- */
-
-#define async_data_read_forward_0_0(exch, method, answer) \
```

Comment:
So where is async_data_read_forward_x_y() and async_data_write_forward_x_y() for 0 <= x < 4 && 0 <= y <= 1?

Reply:
Good question.
I was thinking that since it's such a fringe use function, we don't really need all that API clutter for it (note those macros are currently not used anywhere). I also wanted to get rid of the `_fast` suffix. So I added just the `_4_1` variant thinking we could add more variants later if needed.

Reply:
Yes, it does seem arbitrary. No, it wouldn't be much work. I'll add them.